core-primitives/component-container/src/component_container.rs

```diff
@@ -0,0 +1,78 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Generic component containers.
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxMutex as Mutex;
+
+#[cfg(feature = "std")]
+use std::sync::Mutex;
+
+use crate::atomic_container::AtomicContainer;
+use std::{marker::PhantomData, sync::Arc};
+
+/// Trait to initialize a generic component.
+pub trait ComponentInitializer {
+	type ComponentT;
```

Comment:
The `T` in generic type parameters stands for `Type`

Reply:
I don't mind either way, but I think we should use the T either for Type or Trait but not for both. Otherwise, I think it will get confusing.

(example where we're using it for trait: https://github.com/integritee-network/worker/blob/master/enclave-runtime/src/lib.rs#L79, https://github.com/integritee-network/worker/blob/master/core-primitives/stf-executor/src/executor.rs#L41)

Reply:
Ofc, if there's no better solution, I'm fine with T. 👍

Reply:
In the case you mentioned I also read it as `Type`, because it's used in that context. It never occurred to me that it could be referred to as `Trait` 🤔

Reply:
@mullefel and I decided it's probably best to use `Type` and `Trait` fully written instead of just `T`. That should prevent misinterpretations in the future.

Reply:
I agree

Reply:
I changed it 👍